actionpack/lib/action_controller/metal/rendering.rb

```diff
@@ -4,6 +4,17 @@ module Rendering
 
     RENDER_FORMATS_IN_PRIORITY = [:body, :text, :plain, :html]
 
+    module ClassMethods
+      # Documentation at ActionController::Renderer#render
+      delegate :render, to: :renderer
+
+      # Returns a renderer class (inherited from ActionController::Renderer)
+      # for the controller.
+      def renderer
+        @renderer ||= Renderer.for(self)
```

Comment:
@brainopia what if two two different controllers use this code, but share the same thread and thus the ivar set on the class? When the second controller calls the `renderer` method, the `@renderer` ivar would already exist, except it'd be instantiated with the wrong controller instance and thus be a factory for the wrong controller.